packages/plugins/plugin-build/src/commands/build/index.ts

```diff
@@ -76,6 +83,40 @@ export default class Build extends BaseCommand {
       this.context.cwd,
       this.context.plugins
     );
+    const { project, workspace: cwdWorkspace } = await Project.find(
+      configuration,
+      this.context.cwd
+    );
+
+    if (!cwdWorkspace)
+      throw new WorkspaceRequiredError(project.cwd, this.context.cwd);
+
+    const rootWorkspace = this.all ? project.topLevelWorkspace : cwdWorkspace;
+
+    const rootCandidates = [
+      rootWorkspace,
+      ...(this.buildTargets.length > 0
+        ? rootWorkspace.getRecursiveWorkspaceChildren()
+        : []),
+    ];
+
+    const buildTargetPredicate = (workspace: Workspace) =>
+      this.buildTargets.some(
+        (t) =>
+          micromatch.isMatch(
+            structUtils.stringifyIdent(workspace.locator),
+            t
+          ) ||
+          micromatch.isMatch(
+            workspace.cwd,
+            `${configuration.projectCwd}${path.posix.sep}${t}`
+          )
```

Comment:
This ensures that the current path based approach still works, and extends it to support globbing and multiple parameters.

The same thing can be achieved by 
```sh
> yarn /path/to/project build
```
I'd suggest removing this in an upcoming major.

Reply:
If you decide that the path logic should removed, I'd consider it deprecated from now on. I'd then suggest to _only_ keep the current behavior for path logic: single param without globs. Multiple params and globs would only be supported for project names.

Reply:
I'll pull this down for testing tomorrow or the next day to make sure I grok it.

I'm not keen to remove the path logic, as pathing to separate packages is very useful when combined with things like CODEOWNERS, golang and standardised naming.
I think we could look to deprecate it from a positional arg and put it behind a flag like `-f path/to/project`. Though, if it works well enough checking package names, and then the path we could leave it as both.

If you're migrating from Bazel, path based is really useful.

Though generally I only use `yarn build` in root or in a workspace.

The command _should_ be smart enough for everyone to try and rebuild everything each time. But, that still only scales so far.

> yarn /path/to/project build

This might just run the build script though, and bypass the plugin?

Reply:
I figured the path api was there for a reason, that's why I initially extended it with glob and multiple params support like I did for project names.

Reply:
> > yarn /path/to/project build

> This might just run the build script though, and bypass the plugin?

Based on my testing, it's equivalent to navigating into the folder and calling `yarn build` from there.

```
> yarn build packages/examples/lorem-ipsum
> yarn packages/examples/lorem-ipsum build
> cd packages/examples/lorem-ipsum && yarn build
```
These produce the same results in my testing.